### PR TITLE
Add list_pipeline_schedules for gitlab

### PR DESCRIPTION
**Pull request by fchorney**
Need this endpoint added. Not too sure what the equivalent endpoint for github would be.

### CODE REVIEW

**Review by mattBrzezinski — APPROVED** [review]
LGTM!

File: src/forges/GitLab/pipeline_schedules.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
+@json struct Schedule
```
Comment (christopher-dG) [inline src/forges/GitLab/pipeline_schedules.jl:1]: Name this `PipelineSchedule` perhaps?

**Review by christopher-dG — APPROVED** [review]
LGTM! One comment about the type name but either or is fine.

File: src/forges/GitLab/pipeline_schedules.jl (hunk 1 above)
Comment (fchorney) [reply src/forges/GitLab/pipeline_schedules.jl:1]: yeah I was on the fence about that, but that probably makes more sense

**Review by mattBrzezinski — APPROVED** [review]

**Pull request merged**